### PR TITLE
Feature kafka

### PR DESCRIPTION
**Pull request by HNDaG**

### CODE REVIEW

**Review by OleksandRomaniuk — APPROVED** [review]
LGTM

File: listening-platform/src/main/kotlin/com/nikitahohulia/listeningplatform/service/UserServiceImpl.kt [hunk 1]
```diff
@@ -88,7 +94,9 @@ class UserServiceImpl(
             .flatMap { (user, publisher) ->
                 user.subscriptions.add(publisher.id!!)
                 userRepository.save(user)
-            }.thenReturn(Unit)
+            }
+            .doOnNext { kafkaUserProducer.sendUserUpdatedEventToKafka(it.toResponse().toProto()) }
+            .thenReturn(Unit)
```
Comment (ajax-shulepa-a) [inline listening-platform/src/main/kotlin/com/nikitahohulia/listeningplatform/service/UserServiceImpl.kt:99]: A user is not updated here, but deleted. how do consumers of your event know about it?

Comment (HNDaG) [reply listening-platform/src/main/kotlin/com/nikitahohulia/listeningplatform/service/UserServiceImpl.kt:99]: I don't really understand. I checked the code lines you are referring to. They are used to updating the user, more specifically, updating the user by filling out the set of publisher IDs which the user is subscribed to. So it's an update and not a delete. Or I didn't understand you correctly?

Comment (ajax-shulepa-a) [reply listening-platform/src/main/kotlin/com/nikitahohulia/listeningplatform/service/UserServiceImpl.kt:99]: sorry, my bad

**Review by ajax-shulepa-a — APPROVED** [review]

**Pull request merged**